src/VisualStudio/IntegrationTest/New.IntegrationTests/CSharp/CSharpCodeActions.cs

```diff
@@ -850,17 +850,17 @@ public static void Main()
             static async Task VerifyDiagnosticInErrorListAsync(string expectedSeverity, TestServices testServices, CancellationToken cancellationToken)
             {
                 await testServices.ErrorList.ShowErrorListAsync(cancellationToken);
-                await Task.Delay(TimeSpan.FromSeconds(10));
+                string[] expectedContents = [$"Class1.cs(7, 13): {expectedSeverity} CS0168: The variable 'x' is declared but never used"];
 
-                string[] expectedContents =
-                [
-                    $"Class1.cs(7, 13): {expectedSeverity} CS0168: The variable 'x' is declared but never used",
-                ];
+                while (true)
```

Comment:
I think there's a helper we can use - maybe https://sourceroslyn.io/#Microsoft.VisualStudio.IntegrationTest.Utilities/Helper.cs,95 (not sure if that still exists)

Reply:
Also wonder if it would be good to either assert the cancellation token isn't None, or explicitly use the hang mitigating one